net/sched/p4tc/p4tc_table.c

```diff
@@ -603,26 +709,170 @@ tcf_table_find_byany(struct p4tc_pipeline *pipeline, struct nlattr *name_attr,
 	return ERR_PTR(err);
 }
 
+static int
+tcf_table_init_default_act(struct net *net, struct nlattr **tb,
+			   struct p4tc_table_defact **default_act,
+			   __u16 curr_permissions, struct netlink_ext_ack *extack)
+{
+	int ret;
+
+	if (!tb[P4TC_TABLE_DEFAULT_ACTION] && !tb[P4TC_TABLE_DEFAULT_PERMISSIONS])
+		return 0;
+
+	*default_act = kzalloc(sizeof(**default_act), GFP_KERNEL);
+	if (!(*default_act))
+		return -ENOMEM;
+
+	if (tb[P4TC_TABLE_DEFAULT_PERMISSIONS]) {
+		__u16 *permissions;
+
+		permissions = nla_data(tb[P4TC_TABLE_DEFAULT_PERMISSIONS]);
+		if (*permissions > (1 << 10) - 1) {
```

Comment:
We could define a max macro?

Reply:
Done here: https://github.com/p4tc-dev/linux-p4tc-pub/pull/33/files#diff-feeeb2f6aa78dcb7c3d03d9ca086c03d2b3e2c38896b50419b9cc556725a4866R730